### PR TITLE
testcase/mqtt/itc: use semaphore in place of pthread condition variable

### PR DESCRIPTION
**Pull request by manoj11588612**
Changes in function:
1.use semaphore in MACRO's ITC_MQTT_WAIT_SIGNAL, ITC_MQTT_SEND_SIGNAL
2.use semphore init, deinit in function _itc_mqtt_init, _itc_mqtt_deinit
3.de-initialise handle with NULL in function itc_mqtt_deinit_n_redeinit_client to deallocate memory

Signed-off-by: manoj <manoj.g2@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [ca6e3deabcefee7f851fcfabaaa482cd82228b8f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ca6e3deabcefee7f851fcfabaaa482cd82228b8f] - Code Rule Check OK.

**Comment by samegautam** [comment]
Looks good to me

File: apps/examples/testcase/ta_tc/mqtt/itc/itc_mqtt_main.c [hunk 1]
```diff
@@ -65,13 +60,8 @@ static int _itc_mqtt_init(void)
...
 	g_mqtt_client_handle = NULL;
 
-	int res = pthread_mutex_init(&g_ocfmutex, NULL);
-	if (res != 0) {
-		ITC_MQTT_LOGE;
-		return -1;
-	}
-	res = pthread_cond_init(&g_ocfcond, NULL);
-	if (res != 0) {
+	int ret = sem_init(&g_mqtt_signal, 0, 0);
```
Comment (dr-venkman) [inline apps/examples/testcase/ta_tc/mqtt/itc/itc_mqtt_main.c]: Please confirm whether mqtt utc uses the same variable. If so, maybe rename this variable.

Comment (manoj11588612) [reply apps/examples/testcase/ta_tc/mqtt/itc/itc_mqtt_main.c]: review comment resolved

**Comment by seinfra** [comment]
Target : [1526084e8e874987b0d87fe3514ba717534cfe19] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1526084e8e874987b0d87fe3514ba717534cfe19] - Code Rule Check OK.

**Pull request merged**